pygribjump/src/pygribjump/pygribjump.py

```diff
@@ -144,6 +144,34 @@ def __init__(self, req, ranges, gridHash=None):
         lib.gribjump_new_request(request, c_reqstr, c_ranges, c_ranges_size, c_hash)
         self.__request = ffi.gc(request[0], lib.gribjump_delete_request)
 
+    @classmethod
+    def from_mask(cls, req: dict[str, str], mask: np.ndarray, gridHash: str = None):
+        """
+        Create a request from a boolean mask.
+        The mask is a 1D array of booleans, where True indicates the value should be extracted.
+        """
+
+        m = np.asarray(mask, dtype=bool).ravel()
+
+        if not m.any():
+            raise ValueError("Mask must contain at least one True value")
+
+        padded = np.concatenate(([False], m, [False]))
+        d      = np.diff(padded.astype(int))
+        starts = np.where(d ==  1)[0]
+        ends   = np.where(d == -1)[0]
+        ranges = list(zip(starts, ends))
+
+        return cls(req, ranges, gridHash)
+
+    @classmethod
+    def from_indices(cls, req: dict[str, str], points: np.ndarray, gridHash: str = None):
+        """
```

Comment:
What would you think about changing the parameter name to `indices` and explicitly stating or validating that this is a 1D array (similarly as in the doctoring for `from_mask`)? I'm wondering if it might happen not infrequently that someone might accidentally pass a 2D array here (supplying x/y indices).